### PR TITLE
Refactor PsycItem Component and Introduce usePsycItem Hook

### PR DESCRIPTION
**Pull request by williamslsy**
- Created the `usePsycItem` hook to encapsulate the complex logic
- Removed direct state management and calculation logic from the component.
- Updated the component to use the states and functions provided by `usePsycItem`.
- Improved conditional rendering logic for better readability.
- Extracted smaller, reusable components for overlays and price displays.
- Removed redundant `isAddressesLoading` as a _prop_ from `MintSection `

### CODE REVIEW

File: components/nft-sale-widget/hooks/usePsycItem.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  );
+  const { isPaused } = usePausedSale(item.batchId);
+  const { tokenInformationData } = useReadTokenInformation(item.tokenId);
+  const proof = useFetchProof(address, item.ipfsHash, isPrivateSale);
+
+  const [copyPrice, setCopyPrice] = useState("0.00");
+  const [priceLoading, setPriceLoading] = useState(true);
+  const [isActive, setIsActive] = useState(true);
+
+  useEffect(() => {
```
Comment (gaelicgod) [inline components/nft-sale-widget/hooks/usePsycItem.ts:54]: @williamslsy this seems cleaner

```typescript 
useEffect(() => {
  if (tokenInformationData) {
    const price = isRandom
      ? formatEther(tokenInformationData[0])  // floorPrice
      : formatEther(tokenInformationData[1]); // ceilingPrice

    setCopyPrice(price);
    setPriceLoading(false);
    setIsActive(tokenInformationData[2]);
  }
}, [tokenInformationData, isRandom]);
```

Comment (gaelicgod) [reply components/nft-sale-widget/hooks/usePsycItem.ts:54]: err check if that works ok

Comment (williamslsy) [reply components/nft-sale-widget/hooks/usePsycItem.ts:54]: lol this is not my code 😅

Comment (williamslsy) [reply components/nft-sale-widget/hooks/usePsycItem.ts:54]: yup.. that works

**Pull request merged**